app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -130,28 +132,22 @@ private fun privateModeAdapterItems() = listOf(AdapterItem.PrivateBrowsingDescri
 private fun onboardingAdapterItems(onboardingState: OnboardingState): List<AdapterItem> {
     val items: MutableList<AdapterItem> = mutableListOf(AdapterItem.OnboardingHeader)
 
-    items.addAll(
-        listOf(
-            AdapterItem.OnboardingThemePicker,
-            AdapterItem.OnboardingToolbarPositionPicker,
-        ),
-    )
-    // Customize FxA items based on where we are with the account state:
-    items.addAll(
-        when (onboardingState) {
-            OnboardingState.SignedOutNoAutoSignIn -> {
-                listOf(
-                    AdapterItem.OnboardingManualSignIn,
-                )
+    val onboarding = FxNimbus.features.onboarding.value()
```

Comment:
I don't fully understand when Nimbus updates for experiements but this appears to have 2 risks that I can think of.

1. This never actually returns the experiment because we call this function too early and everyone only sees `default` order
2. Since this runs 3 times its possible that a user starts at `default` order but then quickly switches to an experiment causing a reordering animation on the adapter

@jhugman I'm not sure how our Nimbus updating happens and since this is on an apps first launch I don't think we're guaranteed to have a response yet? Would you be able to clarify?

Reply:
@zmckenney it being fired three times is certainly not ideal, and exactly the kind of review I'm glad I'm getting 🙏 

This feature should only be manipulated using first-run experiments, I'll make it explicit in the FML file!

In a first-run experiment, the feature configuration would be embedded in the app in the [initial_experiments.json](https://github.com/mozilla-mobile/fenix/blob/b2937496792c25aee66c621b69cd8d8e158d458b/app/src/main/res/raw/initial_experiments.json#L4). The `inital_experiments.json` is loaded synchronously at app startup with a [timeout](https://github.com/mozilla-mobile/fenix/blob/b2937496792c25aee66c621b69cd8d8e158d458b/app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt#L64) that means the configuration is available early enough as long as the file loads quick enough. @jhugman might correct me here if I'm wrong 🙈 

If the loading from the file errors for taking too long, then the user wouldn't be enrolled, which is OK either way since the experiment will only target users on their first run.

I'll see if I can move the feature querying so that it's called exactly once (and applied the same way each time the `onboardingAdapterItems` is called!